### ISSUE TITLE
Can't see files copied into container

### ISSUE THREAD

**Issue by bertinatto** [projectatomic/buildah#343]
Hi,

It's not possible to see the files copied into `/var/www/html`. This directory specifically is a volume:

https://github.com/docker-library/wordpress/blob/master/Dockerfile-debian.template#L28

How to reproduce the issue:

```sh
$ sudo ./buildah --debug copy wordpress-working-container-2 https://github.com/projectatomic/buildah/blob/master/README.md /var/www/html/
DEBU[0000] [graphdriver] trying provided driver "overlay"
DEBU[0000] overlay: override_kernelcheck=true
DEBU[0000] backingFs=extfs,  projectQuotaSupported=false
DEBU[0000] saving "https://github.com/projectatomic/buildah/blob/master/README.md" to "/var/lib/containers/storage/overlay/282ba936372f577cbee0be367ac7c14011a2cc3af07dc1e3697515ef5bdb58ff/merged/var/www/html/README.md"

$ sudo ./buildah --debug copy wordpress-working-container-2 https://github.com/projectatomic/buildah/blob/master/README.md /var/www/
DEBU[0000] [graphdriver] trying provided driver "overlay"
DEBU[0000] overlay: override_kernelcheck=true
DEBU[0000] backingFs=extfs,  projectQuotaSupported=false
DEBU[0000] saving "https://github.com/projectatomic/buildah/blob/master/README.md" to "/var/lib/containers/storage/overlay/282ba936372f577cbee0be367ac7c14011a2cc3af07dc1e3697515ef5bdb58ff/merged/var/www/README.md"

$ sudo ./buildah run wordpress-working-container-2 bash
root@mrsdalloway:/var/www/html# ls
.  ..
root@mrsdalloway:/var/www/html# ls /var/www -Z
system_u:object_r:container_file_t:s0:c485,c498 README.md  system_u:object_r:container_file_t:s0:c485,c498 html
```

Some details:

* If I `buildah commit` the container and use `docker run` to check the images's content, I can see the files there.

* This is what I see in `journalctl -f` when I execute `buildah run wordpress-working-container-2 bash`: https://gist.github.com/bertinatto/ab182d85a5eae0edf24778d451d2774c

**Comment by rhatdan** [comment]
We know we have an SELinux issue with buildah 1.8 which prevents writing to the volume.  But I would not expect any content written to the volume to show up in the image?

**Comment by rhatdan** [comment]
We know we have an SELinux issue with buildah 1.8 which prevents writing to the volume.  But I would not expect any content written to the volume to show up in the image?

**Comment by rhatdan** [comment]
@nalind PTAL

**Comment by TomSweeneyRedHat** [comment]
The problem seems to be the VOLUME command in the Dockerfile.  I've shrunk the Dockerfile than @bertinatto sent down a lot and can replicate the issue easily.  This also is replicating #334 easily, I don't know if they're related or not.

```
# cat Dockerfile
FROM fedora 
VOLUME /var/www/html

# buildah bud -t fed .
STEP 1: FROM fedora
Getting image source signatures
Copying blob sha256:a8ee583972c2295bb76704d4defe5116d5e4dd7ba3767aaa2cc8fcf71088ee06
 82.80 MiB / 82.80 MiB [===================================================] 12s
Copying config sha256:422dc563ca3260ad9ef5c47a1c246f5065d7f177ce51f4dd208efd82967ff182
 2.29 KiB / 2.29 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
STEP 2: VOLUME /var/www/html
STEP 3: COMMIT containers-storage:[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.override_kernel_check=true]docker.io/library/fed:latest

# ctrid=$(buildah from fed)
# buildah copy $ctrid temp.txt /var/www/html

# buildah run $ctrid bash
error resolving mountpoints for container: error populating directory "/var/lib/containers/storage/overlay-containers/a760f3e10f2a7b544b66868495e5d9725a9ef5a820a2407c8aa2f4cdd2e04683/userdata/buildah-volumes/69dd2b02e1f3a65918182048ea4e29979a849d8942e8f53ed20a4bf10e529b36" for volume "/var/www/html" in container "a760f3e10f2a7b544b66868495e5d9725a9ef5a820a2407c8aa2f4cdd2e04683" using contents of "/var/lib/containers/storage/overlay/5cc9203b32fab9efef4d0d5f98598bff1e9847903a026c4ceebedb2ec50c0637/merged/var/www/html": Can't copy a directory

# buildah run $ctrid bash
[root@mrsdalloway /]# ls /var/www/html
[root@mrsdalloway /]# exit

# buildah copy $ctrid temp.txt /var/www
# buildah run $ctrid bash
[root@mrsdalloway /]# ls /var/www/html
[root@mrsdalloway /]# ls /var/www     
html  temp.txt
```

I tried creating a Fedora container without the Dockerfile and ran into a perm issue:

```
#buildah from fedora

# buildah run -v "/var/www/html:/var/www/html" fedora-working-container /bin/bash
[root@mrsdalloway /]# ls
bin  boot  dev  etc  home  lib  lib64  lost+found  media  mnt  opt  proc  root  run  sbin  srv  sys  tmp  usr  var
[root@mrsdalloway /]# ls /var/www/html
ls: cannot open directory '/var/www/html': Permission denied
[root@mrsdalloway /]# ls -alF /var/www
total 0
drwxr-xr-x. 3 root root 18 Feb 10 20:35 ./
drwxr-xr-x. 1 root root 17 Feb 10 20:35 ../
drwxr-xr-x. 2 root root  6 Oct 25 12:35 html/
[root@mrsdalloway /]# whoami
root

# buildah copy fedora-working-container temp.txt /var/www/
# buildah run -v "/var/www/html:/var/www/html" fedora-working-container /bin/bash
[root@mrsdalloway /]# ls /var/www
html  temp.txt
```

and I found some lovely avc's in journalctl for the last bit with the perms.

```
Feb 10 15:33:09 localhost.localdomain kernel: SELinux: mount invalid.  Same superblock, different security settings for (dev mqueue, type mqueue)
Feb 10 15:35:47 localhost.localdomain kernel: SELinux: mount invalid.  Same superblock, different security settings for (dev mqueue, type mqueue)
Feb 10 15:35:58 localhost.localdomain audit[30655]: AVC avc:  denied  { read } for  pid=30655 comm="ls" name="html" dev="dm-0" ino=11548535 scontext=system_u:system_r:container_t:s0:c86,c383 tcontext=system_u:object_r:httpd_sys_content_t:s0 tclass=dir permissive=0
Feb 10 15:36:57 localhost.localdomain kernel: SELinux: mount invalid.  Same superblock, different security settings for (dev mqueue, type mqueue)
Feb 10 15:37:00 localhost.localdomain audit[30703]: AVC avc:  denied  { read } for  pid=30703 comm="ls" name="html" dev="dm-0" ino=11548535 scontext=system_u:system_r:container_t:s0:c86,c383 tcontext=system_u:object_r:httpd_sys_content_t:s0 tclass=dir permissive=0
Feb 10 15:37:51 localhost.localdomain kernel: SELinux: mount invalid.  Same superblock, different security settings for (dev mqueue, type mqueue)
Feb 10 15:42:15 localhost.localdomain audit[30757]: AVC avc:  denied  { setattr } for  pid=30757 comm="chown" name="html" dev="dm-0" ino=11548535 scontext=system_u:system_r:container_t:s0:c86,c383 tcontext=system_u:object_r:httpd_sys_content_t:s0 tclass=dir permissive=0

```

I didn't find any AVC's when the image was built using the Dockerfile, so I don't know if it's related or not.

**Comment by bertinatto** [comment]
I followed the steps I reported above and the issue seems to be solved now. I think (but I'm not 100% sure) that #471 by @rhatdan solved it.

Also, I was investigating the error above reported by @TomSweeneyRedHat and I found out that this was actually a `runc` issue. As soon as I updated to the latest version of `runc` (`master` branch) the permission error and the AVC's in `journalctl` were gone.
